app/models/log_event.rb

```diff
@@ -3,8 +3,92 @@ class LogEvent < ActiveRecord::Base
   belongs_to :user
   belongs_to :loggable, polymorphic: true
 
-  def self.log(loggable, event_type, user)
-    create(loggable: loggable, event_type: event_type, user_id: user.id)
+  def self.log(loggable, event_type, event_time, user)
```

Comment:
I feel moderately strongly that we should make the event time an optional keyword argument and get rid of it in most of the calls. I think it really clutters up the API.